staging/src/k8s.io/library-go/README.md

```diff
@@ -0,0 +1,23 @@
+# library-go
+
+This repository provides helpers primarily for core components (core components as described in [Create a k8s.io/component-base repo](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cluster-lifecycle/wgs/0032-create-a-k8s-io-component-repo.md#component-definition)) which are required by at least two components living outside
+of the main `k8s.io/kubernetes` package. Yet, still with a high level of abstraction.
+
+The only allowed kubernetes dependencies are `k8s.io/apimachinery`, `k8s.io/api` and `k8s.io/client-go`.
+
+## Purpose
+
+One of the goals is to provide a better location for helpers currently located under `k8s.io/kubernetes/pkg/apis`. Recent effort of moving [scheduling
+ framework](https://kubernetes.io/docs/concepts/scheduling-eviction/scheduling-framework/) under
+`k8s.io/kube-scheduler` requires duplication of many helper functions
+(see [#91782](https://github.com/kubernetes/kubernetes/issues/91782) for more details).
```

Comment:
@soltysh provide some verbiage for k/kubectl rbac libraries

Reply:
xref:

https://github.com/kubernetes/kubernetes/pull/74879
https://github.com/liggitt/kubernetes/commits/extract-rbac

Reply:
from sig-cli pov:

Another example is shared RBAC code which is blocking extracting kubectl to staging (see https://github.com/kubernetes/enhancements/issues/1020). This problem dates all the way back to December 2018 (see SIG-CLI call from December 19, 2018: https://docs.google.com/document/d/1r0YElcXt6G5mOWxwZiXgGu_X6he3F--wKwg-9UBc29I/edit?pli=1). Recently the topic was touched during sig-auth call (see https://docs.google.com/document/d/1woLGRoONE3EBVx-wTb4pvp4CI7tmLZ6lS26VTbosLKM/edit?ts=5ef3be6a#heading=h.etc9yylhln8x).